src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -3902,3 +3902,97 @@ class DeleteStatementSegment(ansi.DeleteStatementSegment):
             optional=True,
         ),
     )
+
+
+class SetClauseSegment(BaseSegment):
```

Comment:
I'm curious -- do we know how much of this syntax is specific to Postgres? Did you consider adding some of this to the ANSI dialect, and if so, why not add it there instead?

Reply:
I think it's fairly specific. Having a quick look BigQuery, Oracle, Snowflake and  T-SQL don't support the bracketed version, though Coachroach (that we base a lot of our ANSI off of) does.

Think I'd prefer to leave as is in Postgres dialect to be honest.

Reply:
np, just interested in your thoughts!